src/middlewared/middlewared/plugins/network.py

```diff
@@ -584,23 +608,45 @@ def iface_extend(self, iface_state, configs, fake=False):
 
         await self.__save_datastores()
 
-        if data['type'] == 'LINK_AGGREGATION':
-            name = data.get('name')
-            if not name:
-                next_lagg = 0
-                laggs = [
-                    i['int_interface']
-                    for i in await self.middleware.call(
-                        'datastore.query',
-                        'network.interfaces',
-                        [('int_interface', '^', 'lagg')],
-                    )
-                ]
-                while f'lagg{next_lagg}' in laggs:
-                    next_lagg += 1
-                name = f'lagg{next_lagg}'
+        async def get_next(prefix, start=0):
+            number = start
+            ifaces = [
+                i['int_interface']
+                for i in await self.middleware.call(
+                    'datastore.query',
+                    'network.interfaces',
+                    [('int_interface', '^', prefix)],
+                )
+            ]
+            while f'{prefix}{number}' in ifaces:
+                number += 1
+            return f'{prefix}{number}'
 
-            interface_id = None
+        interface_id = None
+        if data['type'] == 'BRIDGE':
+            # For bridge we want to start with 2 because bridge0/bridge1 may have been used
```

Comment:
What if we set up a bridge first and then spin up jail/VM? Can't we have more than one jail/VM bridge?

Reply:
And if we want another jail/VM bridge, won't it auto-create bridge3 (e.g. if we have bridge2 created by middleware)? I mean, we can easily extend this method to check if `bridgeN` also does not exists in the OS, won't it be more reliable?

Reply:
No, only one bridge is used for all of them, 10 VMs do not create 10 bridges.

Reply:
Bridge to which interface? What if I want VM A to have a bridge to `em1` and VM B to have a bridge to `em2`?

Reply:
Bridges are created dynamically by Jails/VM, so we cant know if they are there or not, nor how many will be. Jails in particular can have fixed bridge names, not dynamically created ones (using the next available).

This is something we will have to change on Jails/VMs at a later stage to play well with middleware bridge, given that interface cannot be shared between multiple bridges.